### PR TITLE
added CardComponent

### PR DESCRIPTION
**Pull request by iameru**
if no `data`, it shows option to add.

missing: second style and using own stylesheets instead of tailwind colors

### CODE REVIEW

**Comment by iameru** [comment]
rewrote it. Now is able to swipe. wrote it very similar to #95 as there this swiping is used aswell.

did however change the styling a bit because full height redblock looks bad with the rounded card. see here for my solution. Is this acceptable? 
second

original design should be like this:
2023-03-01-165312_900x371_scrot


now that I think of this, I could also try to make a red background with a different z-index instead, is this wanted?

**Comment by iameru** [comment]
here is the proper solution 
Peek 2023-03-02 11-18

**Comment by iameru** [comment]
there were design changes so back to draft

**Comment by iameru** [comment]
current state:
Peek 2023-03-06 16-21

**Comment by iameru** [comment]
current state
Peek 2023-03-06 17-00

**Comment by iameru** [comment]
Peek 2023-03-07 11-33

**Comment by iameru** [comment]
i hope this is the final design. making PR

File: src/components/Card.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+          basicCardStyle,
+          "-z-10 absolute",
+          cardBackgroundSwipe === "left" && "bg-ux-error",
+          cardBackgroundSwipe === "right" && "bg-ux-success"
+        )}
+        id="swipe-bg"
+      ></div>
+      <SwipeableListItem
+        listType={Type.IOS}
+        trailingActions={trailingActions()}
```
Comment (NoHara42) [inline src/components/Card.tsx:79]: does trailingActions need to be a callable function? 
Couldn't this also just reference a binding?

File: src/components/Card.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    setCardBackgroundSwipe(direction);
+  }
+
+  const basicCardStyle = "rounded-2xl  w-full h-44 ";
+
+  return (
+    <Transition
+      show={swiped}
+      appear={true}
+      enter="transform transition duration-[400ms]"
```
Comment (NoHara42) [inline src/components/Card.tsx:61]: I've seen `duration-[400ms]` often in the project, maybe it's time for an entry in the tailwind config?

**Review by NoHara42 — CHANGES_REQUESTED** [review]
generally looks great, some small amendments then g2g

File: src/components/Card.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        >
+          <div className="flex flex-col gap-3">
+            <p className="text-2xl font-semibold">
+              {checkedItems}/{totalItems} Items
+            </p>
+            {createNewCard ? (
+              <input
+                type="text"
+                placeholder="new list"
+                className="text-4xl uppercase font-heading bg-[transparent] placeholder:text-primary-transparent text-primary-default-Solid focus:outline-none"
```
[the remark this answers is not in the document]
Comment (iameru) [reply src/components/Card.tsx:101]: we don't have `bg-transparent`  in the project as we overwrote the tailwind colorconfig, I guess.

File: src/components/Card.tsx (hunk 2 above)
Comment (iameru) [reply src/components/Card.tsx]: added in 1c1c16d

File: src/components/Card.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (iameru) [reply src/components/Card.tsx]: added in 1c1c16d

File: src/components/Card.tsx (hunk 1 above)
Comment (iameru) [reply src/components/Card.tsx]: refactored in db9c0c6

**Pull request merged**